pkg/util/nodecache.go

```diff
@@ -0,0 +1,115 @@
+package util
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+)
+
+type NodeCache struct {
+	BasePath string
+}
+
+var (
+	CacheSubPath string
```

Comment:
Apart from this introducing race conditions and a lot more (arguably quite bug-prone) complexity, why is this even needed? Why not just store both volume and snapshot metadata under the same directory?